padl/transforms.py

```diff
@@ -91,9 +91,46 @@ def __init__(self, call_info: Optional[inspector.CallInfo] = None,
         self._pd_call_info = call_info
         self._pd_varname = _notset
         self._pd_name = pd_name
-        self._pd_component = {'forward'}
         self._pd_device = 'cpu'
         self._pd_layers = None
+        self._pd_splits = None
+
+    def _pd_get_splits(self, input_components=0) -> Tuple[Union[int, List],
+                                                          Tuple['Transform', 'Transform', 'Transform']]:
+        """ Split the transform into "preprocessing", "forward" and "postprocessing" parts.
+
+        *input_components* contains information about the "split" the input is in. It is either an
+        int or a (potentially nested) list of ints. The ints have the following meaning:
+            - 0 means "not batchified"
+            - 1 means "batchified"
+            - 2 means "unbatchified" a.k.a. post-process
+        If it's a (nested) list, the structure represents the input structure for the transform,
+        whereas the entries represent the "split" of parts of the input.
+
+        # TODO When input_components is a list/tuple of ints does this indicate there are multiple "pipes"?
```

Comment:
@wuhu I'm trying to better understand `input_components`. So I would like to ask When input_components is a list/tuple of ints does this indicate there are multiple "pipes"?

Reply:
Yes, if `input_components` is a list, it indicates multiple pipes. 
However, if it's not a list, it could still be multiple pipes, all in the same "component".

Reply:
I'll remove the TODO and add this info into the doc string.